users/service.go

```diff
@@ -260,15 +274,15 @@ func (svc service) UpdateClientIdentity(ctx context.Context, token, clientID, id
 func (svc service) GenerateResetToken(ctx context.Context, email, host string) error {
 	client, err := svc.clients.RetrieveByIdentity(ctx, email)
 	if err != nil || client.Credentials.Identity == "" {
-		return repoerr.ErrNotFound
+		return svcerr.ErrNotFound
```

Comment:
This can lead to data leak, Any one can check user exists in platform or not, by requesting reset token 
@dborovcanin & @rodneyosodo  What do you think about this approach